File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/global/ClassDeclaration.scala
package io.github.chaosunity.casc.parsing.global

import io.github.chaosunity.casc.parsing.`class`.Function

import java.util

class ClassDeclaration(private var _name: String, private var _methods: util.List[Function]) {
    def name: String = _name

    def methods: util.List[Function] = _methods
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/ClassDeclaration.kt
package io.github.chaosunity.casc.parsing

import io.github.chaosunity.casc.parsing.scope.AccessModifier
import io.github.chaosunity.casc.parsing.scope.Field

data class ClassDeclaration(val name: String, val functions: List<Function<*>>, val fields: List<Field>, val accessModifier: AccessModifier)